internal/beater/beater.go

```diff
@@ -177,6 +177,33 @@ func (s *Runner) Run(ctx context.Context) error {
 		}
 	}
 
+	// Obtain the memory limit for the APM Server process. Certain config
+	// values will be sized according to the maximum memory set for the server.
+	var memLimit float64
+	if cgroupReader := newCgroupReader(); cgroupReader != nil {
+		if limit, err := cgroupMemoryLimit(cgroupReader); err != nil {
+			s.logger.Warn(err)
+		} else {
+			memLimit = float64(limit) / 1024 / 1024 / 1024
```

Comment:
Did you say you were going to limit the use of cgroup memory, to allow for other processes in the container?

Reply:
I capped it to 80% of the cgroup memory limit